implementation/src/main/java/io/smallrye/jwt/config/JWTAuthContextInfoProvider.java

```diff
@@ -238,36 +242,25 @@ private static JWTAuthContextInfoProvider create(String publicKey, String public
             contextInfo.setTokenHeader(tokenHeader);
         }
 
-        if (tokenKeyId != null && tokenKeyId.isPresent()) {
-            contextInfo.setTokenKeyId(tokenKeyId.get());
-        }
-        if (requireNamedPrincipal != null && requireNamedPrincipal.isPresent()) {
-            contextInfo.setRequireNamedPrincipal(requireNamedPrincipal.get());
-        }
+        contextInfo.setTokenKeyId(tokenKeyId.orElse(null));
+        contextInfo.setRequireNamedPrincipal(requireNamedPrincipal.orElse(null));
         SmallryeJwtUtils.setContextTokenCookie(contextInfo, tokenCookie);
-        if (defaultSubClaim != null && defaultSubClaim.isPresent()) {
-            contextInfo.setDefaultSubjectClaim(defaultSubClaim.get());
-        }
+        contextInfo.setDefaultSubjectClaim(defaultSubClaim.orElse(null));
         SmallryeJwtUtils.setContextSubPath(contextInfo, subPath);
-        if (defaultGroupsClaim != null && defaultGroupsClaim.isPresent()) {
-            contextInfo.setDefaultGroupsClaim(defaultGroupsClaim.get());
-        }
+        contextInfo.setDefaultGroupsClaim(defaultGroupsClaim.orElse(null));
         SmallryeJwtUtils.setContextGroupsPath(contextInfo, groupsPath);
-
-        if (expGracePeriodSecs != null && expGracePeriodSecs.isPresent()) {
-            contextInfo.setExpGracePeriodSecs(expGracePeriodSecs.get());
-        }
-
-        if (this.jwksRefreshInterval != null && jwksRefreshInterval.isPresent()) {
-            contextInfo.setJwksRefreshInterval(jwksRefreshInterval.get());
-        }
-
+        contextInfo.setExpGracePeriodSecs(expGracePeriodSecs.orElse(null));
+        contextInfo.setJwksRefreshInterval(jwksRefreshInterval.orElse(null));
         SmallryeJwtUtils.setWhitelistAlgorithms(contextInfo, whitelistAlgorithms);
 
         return Optional.of(contextInfo);
     }
 
     protected void decodeMpJwtPublicKey(JWTAuthContextInfo contextInfo) {
+        if (!mpJwtPublicKey.isPresent() || NONE.equals(mpJwtPublicKey.get())) {
+            return;
+        }
+
```

Comment:
I think all the optimization in this file will have to be reverted, otherwise Thorntail won't be able to use its own properties

Reply:
I am not sure that I follow you here. The main change here is to alter the conditions to assume CDI is working properly and that `Optional` objects are always present (which they should be for MP config properties). That includes setting the default values in the `create` method which is used by unit tests. I took a look at the [class having the same name in Thorntail](https://github.com/thorntail/thorntail/blob/master/fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java) and I don't believe there is an impact since that class provides its own producer method which uses this class's getters.

Thoughts?

Reply:
Sorry, should've clarified - you moved the property checks inside the individual methods, and also introduced many default values but unfortunately it will break Thorntail which has the legacy code such as [this one](https://github.com/thorntail/thorntail/blob/master/fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java#L87).

Reply:
Actually, it does look good, I looked too quickly. (by the way I might've deleted some of your comments by mistake, meant to remove one of my ones).

Reply:
The defaults should all be the same applied during construction of `JWTAuthContextInfo` except for `expGracePeriodSecs`, which should be changed to 60 in the `orElse` call. It looks like the condition you linked in Thorntail is redundant, but it appears to me that it would not break. The Thorntail condition and the one in the SmallRye super class are referencing the same `Optional` value and performing the same check.

Reply:
Ah, I didn't see your "update" comment :-)

Reply:
Sorry :-)